hierarchicalforecast/evaluation.py

```diff
@@ -1,17 +1,231 @@
+    assert (weights is None) or (np.sum(weights) > 0), 'Sum of weights cannot be 0'
+    assert (weights is None) or (weights.shape == y.shape),\
+        f'Wrong weight dimension weights.shape {weights.shape}, y.shape {y.shape}'
+
+def mse(y: np.ndarray, y_hat: np.ndarray, 
+        weights: Optional[np.ndarray] = None,
+        axis: Optional[int] = None) -> Union[float, np.ndarray]:
+    """Mean Squared Error
+
+    Calculates Mean Squared Error between
+    `y` and `y_hat`. MSE measures the relative prediction
+    accuracy of a forecasting method by calculating the 
+    squared deviation of the prediction and the true
+    value at a given time, and averages these devations
+    over the length of the series.
+
+    $$ \mathrm{MSE}(\\mathbf{y}_{\\tau}, \\mathbf{\hat{y}}_{\\tau}) = \\frac{1}{H} \\sum^{t+H}_{\\tau=t+1} (y_{\\tau} - \hat{y}_{\\tau})^{2} $$
+
+    **Parameters:**<br>
+    `y`: numpy array, Actual values.<br>
+    `y_hat`: numpy array, Predicted values.<br>
+    `mask`: numpy array, Specifies date stamps per serie to consider in loss.<br>
+
+    **Returns:**<br>
+    `mse`: numpy array, (single value).
+    """
+    _metric_protections(y, y_hat, weights)
+
+    delta_y = np.square(y - y_hat)
+    if weights is not None:
+        mse = np.average(delta_y[~np.isnan(delta_y)],
+                         weights=weights[~np.isnan(delta_y)],
+                         axis=axis)
+    else:
+        mse = np.nanmean(delta_y, axis=axis)
+    return mse
+
+def mqloss(y: np.ndarray, y_hat: np.ndarray, 
+           quantiles: np.ndarray, 
+           weights: Optional[np.ndarray] = None,
+           axis: Optional[int] = None) -> Union[float, np.ndarray]:
+    """Multi-Quantile Loss
+
+    Calculates the Multi-Quantile loss (MQL) between `y` and `y_hat`.
+    MQL calculates the average multi-quantile Loss for
+    a given set of quantiles, based on the absolute 
+    difference between predicted quantiles and observed values.
+
+    $$ \mathrm{MQL}(\\mathbf{y}_{\\tau},[\\mathbf{\hat{y}}^{(q_{1})}_{\\tau}, ... ,\hat{y}^{(q_{n})}_{\\tau}]) = \\frac{1}{n} \\sum_{q_{i}} \mathrm{QL}(\\mathbf{y}_{\\tau}, \\mathbf{\hat{y}}^{(q_{i})}_{\\tau}) $$
+
+    The limit behavior of MQL allows to measure the accuracy 
+    of a full predictive distribution $\mathbf{\hat{F}}_{\\tau}$ with 
+    the continuous ranked probability score (CRPS). This can be achieved 
+    through a numerical integration technique, that discretizes the quantiles 
+    and treats the CRPS integral with a left Riemann approximation, averaging over 
+    uniformly distanced quantiles.    
+
+    $$ \mathrm{CRPS}(y_{\\tau}, \mathbf{\hat{F}}_{\\tau}) = \int^{1}_{0} \mathrm{QL}(y_{\\tau}, \hat{y}^{(q)}_{\\tau}) dq $$
+
+    **Parameters:**<br>
+    `y`: numpy array, Actual values.<br>
+    `y_hat`: numpy array, Predicted values.<br>
+    `quantiles`: numpy array,(n_quantiles). Quantiles to estimate from the distribution of y.<br>
```

Comment:
Specify quantiles should be in [0,1] interval.

Reply:
Done, added protection
And protection's unit test.